### ISSUE TITLE
[BUG] Blue_H2 and Grey_H2 error iterating over empty sets

### ISSUE THREAD

**Issue by magiovanniello** [macroenergy/Dolphyn.jl#221]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

In write_h2_balance_zone.jl at line 42 and 43, if on generators were designated as Blue_H2 = 1 or Grey_H2 = 1, then there is an error trying to iterate over an empty set.

42. Blue_H2_Generation = sum(sum(inputs["omega"].* value.(EP[:vH2Gen])[y,:] for y in BLUE_H2))
43. Grey_H2_Generation = sum(sum(inputs["omega"].* value.(EP[:vH2Gen])[y,:] for y in GREY_H2))

### Expected Behavior

There should be no attempt to iterate over these (empty) sets. To fix, change to the code below, which is using the exact logic as used later in the file for other variables. 

<img width="815" alt="Screenshot 2024-02-06 at 6 25 46 PM" src="https://github.com/macroenergy/Dolphyn.jl/assets/135251021/926efdab-92d6-422a-8158-dd07fba89d16">

### Environment

```markdown
- OS:MIT Supercloud
- Julia version:9.12
- Solver:Gurobi
- Solver version:10.0
```

### Manifest.toml file

[Manifest.txt](https://github.com/macroenergy/Dolphyn.jl/files/14187110/Manifest.txt)

**Comment by Junwenlaw** [comment]
Yes your edit to the code make sense. Please go ahead and make the edits.

**Comment by RuaridhMacd** [comment]
Resolved by #226
